event_listener/src/main/java/org/jetbrains/maven/server/IntellijMavenSpy.java

```diff
@@ -66,86 +68,83 @@ else if (event instanceof DependencyResolutionResult) {
     catch (Throwable e) {
       collectAndPrintLastLinesForEA(e);
     }*/
-  }
-
-  private static void onDependencyResolutionRequest(DependencyResolutionRequest event) {
-    String projectId = event.getMavenProject() == null ? "unknown" : event.getMavenProject().getId();
-    printMavenEventInfo("DependencyResolutionRequest", "id", projectId);
-  }
-
-  private static void onDependencyResolutionResult(DependencyResolutionResult event) {
-    List<Exception> errors = event.getCollectionErrors();
-    StringBuilder result = new StringBuilder();
-    for (Exception e : errors) {
-      if (result.length() > 0) {
-        result.append(NEWLINE);
-      }
-      result.append(e.getMessage());
     }
-    printMavenEventInfo("DependencyResolutionResult", "error", result);
-  }
-
-  private static void collectAndPrintLastLinesForEA(Throwable e) {
-    //need to collect last 3 lines to send to EA
-    int lines = Math.max(e.getStackTrace().length, 3);
-    StringBuilder builder = new StringBuilder();
-    builder.append(e.getMessage());
-    for (int i = 0; i < lines; i++) {
-      builder.append(e.getStackTrace()[i]).append("\n");
+
+    private static void onDependencyResolutionRequest(DependencyResolutionRequest event) {
+        String projectId = event.getMavenProject() == null ? "unknown" : event.getMavenProject().getId();
+        printMavenEventInfo("DependencyResolutionRequest", "id", projectId);
     }
-    printMavenEventInfo("INTERR", "error", builder);
-  }
-
-  private static void onRepositoryEvent(RepositoryEvent event) {
-    String errMessage = event.getException() == null ? "" : event.getException().getMessage();
-    String path = event.getFile() == null ? "" : event.getFile().getPath();
-    String artifactCoord = event.getArtifact() == null ? "" : event.getArtifact().toString();
-    printMavenEventInfo(event.getType(), "path", path, "artifactCoord", artifactCoord, "error", errMessage);
-  }
-
-  private static void onExecutionEvent(ExecutionEvent event) {
-    MojoExecution mojoExec = event.getMojoExecution();
-    String projectId = event.getProject() == null ? "unknown" : event.getProject().getId();
-    if (mojoExec != null) {
-      String errMessage = event.getException() == null ? "" : getErrorMessage(event.getException());
-      printMavenEventInfo(event.getType(), "source", mojoExec.getSource(), "goal", mojoExec.getGoal(), "id", projectId, "ERR",
-                          errMessage);
+
+    private static void onDependencyResolutionResult(DependencyResolutionResult event) {
+        List<Exception> errors = event.getCollectionErrors();
+        StringBuilder result = new StringBuilder();
+        for (Exception e : errors) {
```

Comment:
this could be done with map as well

Reply:
Big chunk of `IntellijMavenSpy` will be removed via https://github.com/bsp-capstone/maven-bsp/pull/5, thereof resolving this.